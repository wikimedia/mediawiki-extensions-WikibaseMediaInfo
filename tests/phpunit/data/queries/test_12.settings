{
	"title": "Weighted tags without statements",
	"term": "some search term",
	"entitiesForTerm": {
		"some search term": [
			{
				"id": "Q1",
				"score": 1.0,
				"synonyms": [
					"Some thing",
					"Something"
				]
			},
			{
				"id": "Q2",
				"synonyms": [
					"Another thing"
				]
			}
		]
	},
	"userLanguage": "en",
	"settings": {
		"normalizeMultiClauseScores": false,
		"entitiesVariableBoost": true,
		"applyLogisticFunction": false,
		"useSynonyms": false,
		"boost": {
			"statement": {"P1": 0, "P2": 0},
			"weighted_tags": {
				"image.linked.from.wikidata.p18/": 10,
				"image.linked.from.wikidata.p373/": 20,
				"image.linked.from.wikidata.sitelink/": 30
			},
			"descriptions.$language": 0,
			"descriptions.$language.plain": 0,
			"category": 0,
			"category.plain": 0,
			"title": 9,
			"title.plain": 0,
			"redirect.title": 0,
			"redirect.title.plain": 0,
			"text": 0,
			"text.plain": 0,
			"auxiliary_text": 0,
			"auxiliary_text.plain": 0,
			"file_text": 0,
			"file_text.plain": 0,
			"heading.plain": 0,
			"heading": 0,
			"suggest": 0
		}
	}
}
