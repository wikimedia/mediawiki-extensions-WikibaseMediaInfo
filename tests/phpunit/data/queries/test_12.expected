{
    "bool": {
        "must": [
            {
                "bool": {
                    "should": [
                        {
                            "multi_match": {
                                "query": "some search term",
                                "fields": [
                                    "all_near_match^5",
                                    "all_near_match.asciifolding^3.75"
                                ]
                            }
                        },
                        {
                            "bool": {
                                "minimum_should_match": 1,
                                "should": [
                                    {
                                        "dis_max": {
                                            "tie_breaker": 1,
                                            "queries": [
                                                {
                                                    "bool": {
                                                        "minimum_should_match": 0,
                                                        "filter": [
                                                            {
                                                                "multi_match": {
                                                                    "query": "some search term",
                                                                    "fields": [
                                                                        "all",
                                                                        "all.plain"
                                                                    ],
                                                                    "operator": "and"
                                                                }
                                                            }
                                                        ],
                                                        "should": [
                                                            {
                                                                "match": {
                                                                    "title": {
                                                                        "query": "some search term",
                                                                        "boost": 9
                                                                    }
                                                                }
                                                            }
                                                        ]
                                                    }
                                                }
                                            ]
                                        }
                                    },
                                    {
                                        "bool": {
                                            "minimum_should_match": 1,
                                            "should": [
                                                {
                                                    "match": {
                                                        "weighted_tags": {
                                                            "query": "image.linked.from.wikidata.p18\/Q1",
                                                            "boost": 10
                                                        }
                                                    }
                                                },
                                                {
                                                    "match": {
                                                        "weighted_tags": {
                                                            "query": "image.linked.from.wikidata.p373\/Q1",
                                                            "boost": 20
                                                        }
                                                    }
                                                },
                                                {
                                                    "match": {
                                                        "weighted_tags": {
                                                            "query": "image.linked.from.wikidata.sitelink\/Q1",
                                                            "boost": 30
                                                        }
                                                    }
                                                },
                                                {
                                                    "match": {
                                                        "weighted_tags": {
                                                            "query": "image.linked.from.wikidata.p18\/Q2",
                                                            "boost": 5
                                                        }
                                                    }
                                                },
                                                {
                                                    "match": {
                                                        "weighted_tags": {
                                                            "query": "image.linked.from.wikidata.p373\/Q2",
                                                            "boost": 10
                                                        }
                                                    }
                                                },
                                                {
                                                    "match": {
                                                        "weighted_tags": {
                                                            "query": "image.linked.from.wikidata.sitelink\/Q2",
                                                            "boost": 15
                                                        }
                                                    }
                                                }
                                            ]
                                        }
                                    }
                                ]
                            }
                        }
                    ],
                    "minimum_should_match": 1
                }
            }
        ],
        "filter": [
            {
                "terms": {
                    "namespace": [
                        0
                    ]
                }
            }
        ]
    }
}